fili-core/src/main/java/com/yahoo/bard/webservice/table/PhysicalTable.java

```diff
@@ -137,14 +140,19 @@ public synchronized void resetColumns(SegmentMetadata segmentMetadata, Dimension
             workingIntervals.clear();
             for (Map.Entry<String, Set<Interval>> nameIntervals : dimensionIntervals.entrySet()) {
                 String physicalName = nameIntervals.getKey();
-                String apiName = getLogicalColumnName(physicalName);
-                Dimension dimension = dimensionDictionary.findByApiName(apiName);
-                // Schema evolution may lead to unknown dimensions, skip these
-                if (dimension == null) {
-                    continue;
-                }
-                DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(this, dimension);
-                workingIntervals.put(dimensionColumn, nameIntervals.getValue());
+
+                getLogicalColumnNames(physicalName).stream()
+                        .map(dimensionDictionary::findByApiName)
+                        .filter(dimension -> dimension != null)
+                        .forEach(
+                                dimension -> {
+                                    DimensionColumn dimensionColumn = DimensionColumn.addNewDimensionColumn(
```

Comment:
But the operations will cause side-affects right (adding things to existing things)? so if the stream runs in parallel things will get synchronization problem. `forEach` is a reduce operation that allows side-affects I believe. Let me know if I'm wrong or not making sense, thanks!